app/models/user.rb

```diff
@@ -0,0 +1,12 @@
+class User < ApplicationRecord
+  before_save{self.email = email.downcase}
+  validates :name, presence: true,
+    length: {maximum: Settings.validate.user.name.maxlength}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true,
+    length: {maximum: Settings.validate.user.email.maxlength},
+    format: {with: VALID_EMAIL_REGEX}, uniqueness: {case_sensitive: false}
+  has_secure_password
```

Comment:
`has_secure_password` hoạt động như nào?

Reply:
hàm này sẽ thực hiện:
- thêm một trường password_digest dạng một mã băm vào db
- Tạo 2 thuộc tính  ảo là password and password_confirmation khi tạo đối tượng và yêu cầu 2 cái giống nhau
- Chứa 1 phương thức authenticate ,phương thức này trả về 1 đối tượng user khi password chính xác

Reply:
good